go/storage/client/init.go

```diff
@@ -55,25 +54,28 @@ func New(
 	ctx context.Context,
 	namespace common.Namespace,
 	ident *identity.Identity,
-	schedulerBackend scheduler.Backend,
 	registryBackend registry.Backend,
 	runtime registry.RuntimeDescriptorProvider,
-	extraWatcherOpts ...committee.WatcherOption,
 ) (api.Backend, error) {
-	watcherOpts := append(extraWatcherOpts, committee.WithAutomaticEpochTransitions())
-	committeeWatcher, err := committee.NewWatcher(
+	nl, err := nodes.NewRuntimeNodeLookup(
 		ctx,
-		schedulerBackend,
 		registryBackend,
 		namespace,
-		scheduler.KindStorage,
-		watcherOpts...,
 	)
 	if err != nil {
-		return nil, fmt.Errorf("storage/client: failed to create committee watcher: %w", err)
+		return nil, fmt.Errorf("storage/client: failed to create runtime node watcher: %w", err)
 	}
 
-	return NewForNodes(ctx, ident, committeeWatcher.Nodes(), runtime)
+	publicStorageNl := nodes.NewFilteredNodeLookup(nl,
+		nodes.WithAllFilters(
+			// Ignore self.
+			nodes.IgnoreNodeFilter(ident.NodeSigner.Public()),
+			// Only storage nodes that opted into public RPC.
+			nodes.TagFilter(nodes.TagsForRoleMask(node.RoleStorageRPC)[0]),
```

Comment:
So if there's no nodes with `RoleStorageRPC` enabled, this won't select any nodes and the returned client will fail all the requests - even for the methods that don't have auth (getting checkpoints and diffs).

I think this filter should be skipped.

Reply:
Note that the this default storage client is specifically used by a client node as the storage and compute workers use their own clients. So I think this is fine?

Reply:
Yeah probably fine then. Should maybe rename the method to something that better indicates this is connecting to public storage nodes specifically.

Reply:
Yeah good idea.